command-not-found.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+# for bash 4
+# this will be called when a command is entered
+# but not found in the user’s path + environment
+command_not_found_handle () {
+    toplevel=nixos # TODO: detect this somehow
+    cmd=$1 # TODO: differentiate between paths and commands
+    attrs=$(@out@/bin/nix-locate -1 --no-group --type x --top-level --regex "/bin/$cmd$")
```

Comment:
you probably don't want to use `--regex` here, as `$cmd` may contain regex special characters (like `[` or `.`). I think `--whole "/bin/$cmd"` should be enough. Although I now notice that `--whole` is probably implemented wrongly. I think right now, it'll match `libexec/bin/$cmd` for example as well, but it really shouldn't.